services/webhook/deliver.go

```diff
@@ -145,6 +147,20 @@ func Deliver(ctx context.Context, t *webhook_model.HookTask) error {
 		Headers: map[string]string{},
 	}
 
+	if w.Type == webhook_model.GITEA {
+		meta := GetGiteaHook(w, secret.DecryptSecret)
+		if meta.AuthHeaderEnabled {
+			var content string
+			switch meta.AuthHeader.Type {
+			case webhook_model.BASICAUTH:
+				content = fmt.Sprintf("Basic %s", base.BasicAuthEncode(meta.AuthHeader.Username, meta.AuthHeader.Password))
+			case webhook_model.TOKENAUTH:
+				content = fmt.Sprintf("token %s", meta.AuthHeader.Token)
```

Comment:
Should it be up to the admin if and what is prepended?

Reply:
What would be the use-case for that?

Reply:
Sometimes the token must be provided with "token <value>" or "Bearer <value>". Right now the second one is not possible.

Reply:
Instead of handling this in the backend, maybe it could be exclusively handled in the frontend:
- if the user chooses "basic auth", the frontend computes the base64 and sends `Basic <computedbase64>`
- if the user chooses "bearer token", the frontend sends `Bearer <token>`
- and maybe a 3rd option for the use to exactly choose the header value

This way the backend logic can be simplified a bit, while keeping maximum flexibility.

Reply:
Yeah, I also thought about frontend code for it. But IIRC, it is preferred to do most logical code in the backend. So I went that way. 🙂

Reply:
That `token` part definitely needs to be configurable, it's called the "schema" IIRC. Many APIs take `Bearer` but in the future, other schema could be introduced.